src/SixLabors.Fonts/TextLayout.cs

```diff
@@ -565,11 +565,11 @@ private static void SubstituteBidiMirrors(FontMetrics fontMetrics, GlyphSubstitu
                     {
                         glyphAdvance *= options.TabWidth;
                     }
-                    else if (CodePoint.IsZeroWidthJoiner(codePoint) || CodePoint.IsZeroWidthNonJoiner(codePoint))
+                    else if (metrics.Length == 1 && (CodePoint.IsZeroWidthJoiner(codePoint) || CodePoint.IsZeroWidthNonJoiner(codePoint)))
                     {
                         // The zero-width joiner characters should be ignored when determining word or
-                        // line break boundaries so are safe to skip here.
-                        // Any existing instances are the result of font error.
+                        // line break boundaries so are safe to skip here. Any existing instances are the result of font error.
+                        // It multiple metrics are associated with code point, they are most likely the result of a substitution so we shouldn't ignore it.
```

Comment:
Gotta love when name and behavior doesn't match eh? 

This fix looks good to me. As far as unit testing goes, assuming we cannot use the Mozilla Font, we'll have to create a Windows only test and reference the font from the system collection. I'm not sure we can legally download and include it. [Distribution](https://docs.microsoft.com/en-us/typography/font-list/segoe-ui-emoji#products-that-supply-this-font) covered 10 and 11 so the CI images should be good.

You can wrap the test in the following compiler conditional `#if OS_WINDOWS`

Reply:
It's not exactly the same use but in the FAQ it says:
>  Web fonts are fonts that are hosted on a web server. You do not have rights to:
>  -  copy fonts from a Windows installation to a web server, a process known as web font “self-hosting”.
>  - convert the font to the formats typically associated with web fonts, such as the WOFF or WOFF2 format.

> You can wrap the test in the following compiler conditional #if OS_WINDOWS

The problem is that the font is very different between Windows 10 and Windows 11. I just tested and some emojis are broken with the old font. PR incoming 😅

Reply:
Thanks for your help! Maybe we can get away with it then, the language appears to allow our use case. If anyone complains we can take it down

Reply:
> If anyone complains we can take it down

I like your pragmatism! 😉

Reply:
I added a simple unit test. I noticed that the width is not exactly the same. I'm not sure if it's the font or if there's still an issue on how the width is computed.